dbt/adapters/base/impl.py

```diff
@@ -1574,13 +1579,24 @@ def valid_incremental_strategies(self):
 
     def builtin_incremental_strategies(self):
         builtin_strategies = ["append", "delete+insert", "merge", "insert_overwrite"]
-        if os.environ.get("DBT_EXPERIMENTAL_MICROBATCH"):
+        if self.behavior.require_batched_execution_for_custom_microbatch_strategy:
             builtin_strategies.append("microbatch")
 
         return builtin_strategies
 
     @available.parse_none
     def get_incremental_strategy_macro(self, model_context, strategy: str):
+        """Gets the macro for the given incremental strategy.
```

Comment:
First, nice docs. We need to add these more often.

Per @jtcohen6's observation, at this point, we know what the user is trying to do and we know what the value of the behavior flag is. What if instead of adding `microbatch` in `builtin_incremental_strategies`, we add it conditionally here:
```python
    # validate strategies for this adapter
    valid_strategies = self.valid_incremental_strategies()
    valid_strategies.append("default")
    builtin_strategies = self.builtin_incremental_strategies()
    if strategy == "microbatch" and self.behavior.require_batched_execution_for_custom_microbatch_strategy:
        builtin_strategies.append("microbatch")
    if strategy in builtin_strategies and strategy not in valid_strategies:
        ...
```
Then the warning should only fire when the user has specified `microbatch`, but has not enabled the built-in microbatch strategy.